xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/exceptions/ExceptionExceptionMapper.java

```diff
@@ -0,0 +1,117 @@
+ * {@link RuntimeException} or already mapped by another exception mapper.
+ * <p>
+ * The goal is to be able to have access to the stacktrace of the exceptions, instead of a simple http 500 error and a
+ * {@code "No ExceptionMapper was found, but must be found"} message (see
+ * <a href="https://github.com/restlet/restlet-framework-java/issues/1070">
+ * https://github.com/restlet/restlet-framework-java/issues/1070</a>).
+ *
+ * @version $Id$
+ * @since 13.8RC1
+ * @since 13.7.1
+ * @since 13.4.4
+ */
+@Component
+@Named("org.xwiki.rest.internal.exceptions.ExceptionExceptionMapper")
+@Provider
+@Singleton
+public class ExceptionExceptionMapper implements ExceptionMapper<Exception>, XWikiRestComponent
+{
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private TemplateManager templateManager;
+
+    @Inject
+    private ScriptContextManager scriptContextManager;
+
+    @Inject
+    private ContextualLocalizationManager contextLocalization;
+
+    @Override
+    public Response toResponse(Exception cause)
+    {
+        // This ExceptionMapper captures too many exception types, because we can't distinguish checked and unchecked
+        // exceptions using generic types. If the mapped type is RuntimeException (unchecked exception), JaxRsProviders
+        // would re-throw the exception without mapping it (or it would have been handled by another mapper instead of
+        // this one), we reproduce this behavior to stay as close as possible to the default implementation (note:
+        // RuntimeException stacktraces are logged by default).
+        if (cause instanceof RuntimeException) {
+            throw (RuntimeException) cause;
+        }
+
+        // Logs the unmapped exception, with its stacktrace.
+        this.logger.error("A REST endpoint failed with an unmapped exception.", cause);
+
+        // Returns an error response to the client, with some detail about the exception in the response.
+        String message;
+        String templateName = "rest/exception.vm";
+        MediaType textPlainType;
+        try {
+            ScriptContext scriptContext = this.scriptContextManager.getScriptContext();
+            scriptContext.setAttribute("cause", cause, ScriptContext.ENGINE_SCOPE);
+            // Make the media type html by default.
+            textPlainType = MediaType.TEXT_HTML_TYPE;
+            message = this.templateManager.render(templateName);
+        } catch (Exception e) {
+            this.logger.warn("Failed to render the response using template [{}]. Cause: [{}].", templateName,
+                getRootCauseMessage(e));
+            // Fallback to a formatted string when the template rendering fails.
+            // We use a hardcoded \n because the message is used in an HTTP response.
+            String translation =
+                this.contextLocalization.getTranslationPlain("rest.exception.noMapper", getRootCauseMessage(cause));
```

Comment:
Given the error message, I would expect the parameter to be the exception class name and not the root cause message (which is printer later in the stack trace).